src/system/MailerModule/Controller/AdminController.php

```diff
@@ -13,85 +13,181 @@
 
 namespace Zikula\MailerModule\Controller;
 
-use Symfony\Component\Security\Core\Exception\AccessDeniedException;
-use Zikula_View;
-use SecurityUtil;
-use FormUtil;
-use Zikula\MailerModule\Form\Handler\ModifyConfigHandler;
-use Zikula\MailerModule\Form\Handler\TestConfigHandler;
-use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\HttpFoundation\RedirectResponse;
-use Symfony\Component\Routing\RouterInterface;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Zikula\Core\Controller\AbstractController;
+use Zikula\ThemeModule\Engine\Annotation\Theme;
 
 /**
+ * Class AdminController
+ * @package Zikula\MailerModule\Controller
  * @Route("/admin")
  */
-class AdminController extends \Zikula_AbstractController
+class AdminController extends AbstractController
 {
     /**
-     * Post initialise.
-     *
-     * @return void
-     */
-    protected function postInitialize()
-    {
-        // In this controller we do not want caching.
-        $this->view->setCaching(Zikula_View::CACHE_DISABLED);
-    }
-
-    /**
      * @Route("")
-     *
-     * the main administration function
-     *
-     * @return RedirectResponse
+     * @deprecated remove at Core-2.0
      */
     public function indexAction()
     {
-        // Security check will be done in modifyconfig()
-        return new RedirectResponse($this->get('router')->generate('zikulamailermodule_admin_modifyconfig', array(), RouterInterface::ABSOLUTE_URL));
+        @trigger_error('The zikulamailermodule_admin_index route is deprecated. please use zikulamailermodule_admin_config instead.', E_USER_DEPRECATED);
+
+        return $this->redirect($this->generateUrl('zikulamailermodule_admin_config'));
     }
 
     /**
      * @Route("/config")
+     * @Theme("admin")
+     * @Template
      *
-     * This is a standard function to modify the configuration parameters of the
-     * module
-     *
-     * @return mixed False on errors, true on redirects, and otherwise it returns the HTML output for the page.
-     *
+     * @param Request $request
      * @throws AccessDeniedException Thrown if the user doesn't have admin access to the module
+     * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function modifyconfigAction()
+    public function configAction(Request $request)
     {
-        // security check
-        if (!SecurityUtil::checkPermission('ZikulaMailerModule::', '::', ACCESS_ADMIN)) {
+        if (!$this->hasPermission('ZikulaMailerModule::', '::', ACCESS_ADMIN)) {
             throw new AccessDeniedException();
         }
 
-        $form = FormUtil::newForm('ZikulaMailerModule', $this);
+        $form = $this->createForm('Zikula\MailerModule\Form\Type\ConfigType');
+
+        if ($form->handleRequest($request)->isValid()) {
```

Comment:
I read that a good while ago in Symfony docs, but can't find it again. If you do not need to perform additional actions between `handleRequest` and `isValid` this is fine.